src/steps/build-team-to-member-mapping/index.ts

```diff
@@ -29,34 +29,89 @@ const step: IntegrationStep<HerokuIntegrationConfig> = {
     },
   ],
   dependsOn: [TEAM_STEP, MEMBER_STEP],
-  async executionHandler({ instance, jobState }) {
-    const heroku = new HerokuClient(instance.config);
+  async executionHandler(context) {
+    const { instance, jobState, logger } = context;
 
+    const heroku = new HerokuClient(instance.config);
     const userIdMap = await createUserIdMap(jobState);
 
-    await jobState.iterateEntities({ _type: TEAM_TYPE }, async (team) => {
-      const teamMembers = await heroku.getTeamMembers(team.id as string);
+    await jobState.iterateEntities(
+      {
+        _type: TEAM_TYPE,
+      },
+      async (teamEntity) => {
+        logger.info(
+          {
+            teamName: teamEntity.name,
+            teamId: teamEntity.id,
+          },
+          'Attempting to fetch team members for team',
+        );
+
+        try {
+          const teamMembers = await heroku.getTeamMembers(
+            teamEntity.id as string,
+          );
 
-      for (const teamMember of teamMembers) {
-        const userId = teamMember.user ? teamMember.user.id : null;
-        const user = userIdMap.get(userId);
+          logger.info(
+            {
+              numTeamMembers: teamMembers.length,
+            },
+            'Successfully fetched team members',
+          );
 
-        if (user) {
-          await jobState.addRelationships([
-            createDirectRelationship({
-              _class: RelationshipClass.HAS,
-              from: team,
-              to: user,
-              properties: { role: teamMember.role },
-            }),
-          ]);
+          await createTeamHasUserRelationships({
+            userIdMap,
+            teamMembers,
+            teamEntity: teamEntity,
+            jobState,
+          });
+        } catch (err) {
+          if (err.code === 'PROVIDER_AUTHORIZATION_ERROR') {
+            logger.publishEvent({
+              name: 'missing_scope',
+              description: `Could not fetch team members. Missing required OAuth scope (endpoint=${err.endpoint}, scope=global)`,
```

Comment:
This API requires a `global` scopes (from what my testing can tell). The `global` scope allows for write permissions and the ability to read secrets. It should be opt-in.

Reply:
It seems we may want to consider collecting these errs across teams and produce a sing job log event instead of the same message multiple times, once for each team.

Reply:
Ah good call. This was an oversight on my part. Will fix this.